Tests/RegexBuilderTests/AlgorithmsTests.swift

```diff
@@ -104,4 +104,38 @@ class RegexConsumerTests: XCTestCase {
        result: "9+16, 3, 10, 99+1")
     )
   }
+
+  func testSwitches() {
+    switch "abcde" {
+    case Regex {
+      "a"
+      ZeroOrMore(.any)
+      "f"
+    }:
+      XCTFail()
+
+    case "abc":
+      XCTFail()
+
+    case Regex {
+      "a"
+      "b"
+      "c"
+    }:
+      XCTFail()
+
+    case Regex {
+      "a"
+      ZeroOrMore(.any)
+      "e"
+    }:
+      break // success
+
+    case OneOrMore(.word):
+      break // success
```

Comment:
Should these be in two different tests? If this is  just to test compilation, no worries.

Reply:
Good point. Fixed.